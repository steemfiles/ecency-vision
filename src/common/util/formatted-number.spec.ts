import formattedNumber from "./formatted-number";

it("(1) formattedNumber", () => {
  const res = formattedNumber(1250);
  expect(res).toMatchSnapshot();
  
   
});

it("(2) formattedNumber - with fraction digits", () => {
  const opts = {
    value: 1250,
    fractionDigits: 2,
  };

  const res = formattedNumber(1250, opts);
  expect(res).toMatchSnapshot();
});

it("(3) formattedNumber - with prefix", () => {
  const opts = {
    prefix: "$",
  };

  const res = formattedNumber(100, opts);
  expect(res).toMatchSnapshot();
});

it("(4) formattedNumber - with suffix", () => {
  const opts = {
    suffix: "TL",
  };

  const res = formattedNumber(100, opts);
  expect(res).toMatchSnapshot();
});

it("(5) formattedNumber - with fraction digits and maximum fraction digits", 
  () => {
  const res1 = formattedNumber(1250, {maximumFractionDigits:3, fractionDigits:1});
  expect(res1).toMatchSnapshot();
  
  const res2 = formattedNumber(1250, {maximumFractionDigits:6, fractionDigits:6});
  expect(res2).toMatchSnapshot();

  const res3 = formattedNumber(1e-7, {maximumFractionDigits:8, fractionDigits:6});
  expect(res3).toMatchSnapshot();
  
  const res4 = formattedNumber("0.0000001", {maximumFractionDigits:8, fractionDigits:6});
  expect(res4).toMatchSnapshot();
  
})