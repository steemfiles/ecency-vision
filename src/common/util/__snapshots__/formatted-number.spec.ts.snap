// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(1) formattedNumber 1`] = `"1,250.000"`;


exports[`(2) formattedNumber - with fraction digits 1`] = `"1,250.00"`;

exports[`(3) formattedNumber - with prefix 1`] = `"$ 100.000"`;

exports[`(4) formattedNumber - with suffix 1`] = `"100.000 TL"`;

exports[`(5) formattedNumber - with fraction digits and maximum fraction digits 1`] = `"1,250.0"`;
exports[`(5) formattedNumber - with fraction digits and maximum fraction digits 2`] = `"1,250.000,000"`;

exports[`(5) formattedNumber - with fraction digits and maximum fraction digits 3`] = `"0.000,000,1"`;
exports[`(5) formattedNumber - with fraction digits and maximum fraction digits 4`] = `"0.000,000,1"`;
